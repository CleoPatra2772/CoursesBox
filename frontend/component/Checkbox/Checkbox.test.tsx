import  userEvent  from "@testing-library/user-event";
import { render, screen } from "@/test-utils";
import { Checkbox } from "./Checkbox";

describe("Checkbox test cases", () => {
    it("Render check", () => {
        const onChange = jest.fn();
        jest.spyOn(Math, "random").mockReturnValue(0.9999999999999);

        const { asFragment } = render(<Checkbox onChange={onChange} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("Check onChange Callback", async () => {
        const onChange = jest.fn();

        render(<Checkbox onChange={onChange} />);

        const element = screen.getByRole("checkbox", {hidden: true});

        await userEvent.click(element);

        expect(onChange).toHaveBeenCalled();
    })
})