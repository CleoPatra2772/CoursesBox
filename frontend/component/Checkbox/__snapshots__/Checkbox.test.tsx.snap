// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test cases Render check 1`] = `
<DocumentFragment>
  <label
    class="css-1p4tilc"
  >
    <input
      id="prefix-ffffffffffe3d8"
      type="checkbox"
    />
    <label
      class="css-1xuc9uu"
      for="prefix-ffffffffffe3d8"
    >
      ✔️
    </label>
  </label>
</DocumentFragment>
`;
